ceph-releases/luminous/ubuntu/16.04/daemon/exit.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# NOTE (leseb): I suspect that leaving the mountpoint opened
+# after closing the namespace might cause some corruption.
+# Especially in the context of encrypted OSDs.
+#
+# The idea here is to stop the osd AND unmount the partition(s)
+# This is not easy since by design the container will exit
+# if PID 1 stops.
+#
+# Ideally, during its shutdown process, the OSD will do that.
+
+set -e
+
+function umount_ceph_mnt {
+  local ceph_mnt
+  ceph_mnt=$(findmnt --nofsroot --noheadings --output SOURCE --submounts --target /var/lib/ceph/osd/ | tail -n +2)
+  for mnt in $ceph_mnt; do
+    log "Unmounting $mnt"
+    umount "$mnt"
+  done
+}
+
+function graceful_stop {
+  echo "Sending SIGTERM to PID $child_for_exec"
+  kill -SIGTERM "$child_for_exec"
+  wait "$child_for_exec"
+
+  umount_ceph_mnt
+  exit 143 # 128 + 15 -- SIGTERM
+}
+
+child_for_exec=1
+
+function exec {
```

Comment:
it could be nice to not declare an override to `exec` bash built-in function a second time.
(probably doing something nicer to reuse [this](https://github.com/ceph/ceph-container/blob/master/ceph-releases/luminous/ubuntu/16.04/daemon/docker_exec.sh#L45))